plugins/scaffolder-backend/src/scaffolder/tasks/NunjucksWorkflowRunner.ts

```diff
@@ -397,10 +389,9 @@ export class NunjucksWorkflowRunner implements WorkflowRunner {
           logger: loggerToWinstonLogger(taskLogger),
           logStream: streamLogger,
           workspacePath,
-          async checkpoint<T extends JsonValue | void>(opts: {
-            key?: string;
-            fn: () => Promise<T> | T;
-          }) {
+          async checkpoint<T extends JsonValue | void>(
+            opts: CheckpointContext<T>,
```

Comment:
Wish this could be `ctx` here. Might need help wording types if this needs to be `CheckpointOptions` as I used that elsewhere.